src/api/app/models/staging/staging_project.rb

```diff
@@ -0,0 +1,141 @@
+module Staging
+  class StagingProject < Project
+    has_many :staged_requests, class_name: 'BsRequest', foreign_key: :staging_project_id, dependent: :nullify
+    has_many :status_reports, through: :repositories, inverse_of: :checkable
+    belongs_to :staging_workflow
+
+    def staging_identifier
+      name[/.*:Staging:(.*)/, 1]
+    end
+
+    def untracked_requests
+      requests_to_review - staged_requests
+    end
+
+    # The difference between staged requests and requests to review is that staged requests are assigned to the staging project.
+    # The requests to review are requests which are not related to the staging project (unless they are also staged).
+    # They simply need a review from the maintainers of the staging project.
+    def requests_to_review
+      @requests_to_review ||= BsRequest.with_open_reviews_for(by_project: name)
+    end
+
+    def building_repositories
+      set_buildinfo unless @building_repositories
+      @building_repositories
+    end
+
+    def broken_packages
+      set_buildinfo unless @broken_packages
+      @broken_packages
+    end
+
+    def missing_reviews
+      return @missing_reviews if @missing_reviews
+
+      @missing_reviews = []
+      attribs = [:by_group, :by_user, :by_project, :by_package]
+
+      (requests_to_review + staged_requests).uniq.each do |request|
+        request.reviews.each do |review|
+          next if review.state == :accepted || review.by_project == name
+          # FIXME: this loop (and the inner if) would not be needed
+          # if every review only has one valid by_xxx.
+          # I'm keeping it to mimic the python implementation.
```

Comment:
@bgeuken 

> That method is just copied over from the old staging project code. We kept it for reference.

I though we don't want just to copy code over but to adapt it. I would write this properly, with a `where` instead of keeping it coherent with some Python code we don't even know where it is and if still keep looking the same. [as said before I would agree to this being done in another PR to get this merged]